gen/calc.py

```diff
@@ -421,6 +421,64 @@ def calculate_cluster_docker_registry_enabled(cluster_docker_registry_url):
     return 'false' if cluster_docker_registry_url == '' else 'true'
 
 
+def validate_cosmos_config(cosmos_config):
```

Comment:
What is `cosmos_config`? I am sure how this symbol is getting resolved. I notice an increasing correlation between the name of the parameter and the name of the property in `entry`. Is this how it is resolved?